### PR TITLE
Restructure copy to blobs

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Restructure location of language specific files

### CODE REVIEW

File: eng/common/scripts/copy-docs-to-blobstorage.ps1
Comment (weshaggard) [inline eng/common/scripts/copy-docs-to-blobstorage.ps1:11]: Should we remove the $Language parameter? Or at least stop passing it in the template?

Comment (chidozieononiwu) [reply eng/common/scripts/copy-docs-to-blobstorage.ps1:11]: I don't understand what you mean.

**Comment by weshaggard** [comment]
I'm fine to do some testing of this but I'd like to wait until after release week before we merge this change in.

File: eng/common/scripts/copy-docs-to-blobstorage.ps1
Comment (weshaggard) [reply eng/common/scripts/copy-docs-to-blobstorage.ps1:11]: This script takes a Language parameter but now Language is setup and coming from common.ps1 and in fact we might not even need to Language in this script any more since we are using the function we define in the language settings.

Comment (chidozieononiwu) [reply eng/common/scripts/copy-docs-to-blobstorage.ps1:11]: Ok, I see what you mean, I'll do that.

**Comment by chidozieononiwu** [comment]
> I'm fine to do some testing of this but I'd like to wait until after release week before we merge this change in.

I was going to do template releases off the Sync PRs.

File: eng/common/scripts/copy-docs-to-blobstorage.ps1
Comment (weshaggard) [inline eng/common/scripts/copy-docs-to-blobstorage.ps1:14]: Looks like a bad merge removed this import.

Comment (weshaggard) [reply eng/common/scripts/copy-docs-to-blobstorage.ps1:14]: Nevermind you imported it in common.

File: eng/common/scripts/copy-docs-to-blobstorage.ps1
Comment (weshaggard) [reply eng/common/scripts/copy-docs-to-blobstorage.ps1:11]: What is your plan for Language? Do you plan to just let common.ps1 override it for now and then remove it from the script once you can update the language repos to stop passing it?

**Comment by ahsonkhan** [comment]
Any updates to this?

**Comment by weshaggard** [comment]
@chidozieononiwu the changes look complete it is just a matter of whether or not you finished your testing.

**Comment by weshaggard** [comment]
@mitchdenny @chidozieononiwu looks like the VerifyAndMerge stage timeout. We might need to use a custom environment for this pipeline and remove the 3d timeout https://dev.azure.com/azure-sdk/internal/_environments/26/checks.

**Comment by weshaggard** [comment]
In the mean time you can run the script locally to merge the Sync PRs when ready.

**Comment by mitchdenny** [comment]
> @mitchdenny @chidozieononiwu looks like the VerifyAndMerge stage timeout. We might need to use a custom environment for this pipeline and remove the 3d timeout https://dev.azure.com/azure-sdk/internal/_environments/26/checks.

Setup the _githibmerges_ environment with the same approver set, but a 30 day timeout (looks like they've gotten rid of infinite timeouts actually.

**Comment by ahsonkhan** [comment]
Any plans for this PR? It's been open in corresponding language repos for 2+ weeks now.

If it is not being worked on actively, consider closing it.

**Comment by chidozieononiwu** [comment]
> Any plans for this PR? It's been open in corresponding language repos for 2+ weeks now.
> 
> If it is not being worked on actively, consider closing it.
> 
> For example, currently, 6 out of 16 PRs in C are sync:
> https://github.com/Azure/azure-sdk-for-c/pulls
> 
> cc @RickWinter

I am still working on it. But other changes need to go in before this change can be merged. This is due to the fact we are still working on how to properly test `eng/common` changes across the language repos.

**Comment by ahsonkhan** [comment]
> I am still working on it. But other changes need to go in before this change can be merged.

In that case, does it make sense to temporarily close this PR, while it is blocked, and then re-open it once it is ready? Or are you finding some value in keeping it open?

**Comment by mitchdenny** [comment]
Hey @ahsonkhan, one of the problems with cross cutting EngSys changes (particularly ones that could break pipelines) is that they take longer to implement. The PRs need to be open so that the changes can be validated.

File: eng/common/scripts/copy-docs-to-blobstorage.ps1
Comment (weshaggard) [inline eng/common/scripts/copy-docs-to-blobstorage.ps1:235]: I assume all the language repo's have been updated already with any changes made to these deleted lines?

Comment (chidozieononiwu) [reply eng/common/scripts/copy-docs-to-blobstorage.ps1:235]: Yes, I did another check yesterday to make sure. And there has been no changes since then.

File: eng/common/pipelines/templates/steps/publish-blobs.yml
Comment (weshaggard) [reply eng/common/pipelines/templates/steps/publish-blobs.yml:24]: OK it would be nice to to file an issue to clean this up.

File: eng/common/scripts/copy-docs-to-blobstorage.ps1
Comment (weshaggard) [inline eng/common/scripts/copy-docs-to-blobstorage.ps1:233]: It might be worth testing to check that this function exists and if not give a nice error message. That will help on-boarding of new languages.

File: eng/common/scripts/copy-docs-to-blobstorage.ps1
Comment (weshaggard) [inline eng/common/scripts/copy-docs-to-blobstorage.ps1:234]: We blindly set the name https://github.com/Azure/azure-sdk-tools/blob/master/eng/common/scripts/common.ps1#L33 so I'd expect that this check will always pass. What we need to do is see if the function exists.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**
